src/c20_server/job_manager.py

```diff
@@ -26,9 +28,13 @@ def report_failure(self, user):
         job = self.in_progress_jobs.unassign(user.user_id)
         self.job_queue.add_job(job)
 
-    @staticmethod
-    def reset_stale_job():
-        return
+    def reset_stale_job(self, time_to_expire):
+        for user in self.r_database.hgetall('assigned_jobs'):
+            job_info = self.r_database.hget('assigned_jobs', user)
+            time_, job = pickle.loads(job_info)
+            if time_ < time_to_expire:
```

Comment:
Yes Ben, what the function does is to unassign ALL jobs that are in the in-progress queue after a certain time is met. And a great point of view of how this function is implemented. But I think this all related to some questions about this! what is the amount time allotted for a job before becoming a stale job? are we going to set the x amount (of hours in seconds) in the flask server(as an optional or default amount will be set)? so we easily could change the amount time allotted if we decided to. If so, how we would deal with that if we add a third entry when it is assigned within inprogress.py? Are we going to call reset_stale_job every time we want to check what is the stale jobs are and unassign them? or instead, it will be looping forever to do the check when we start the server? 
I would say we as group should come up with answers for these questions, so we can have more understand about how the server should work and run.